ui/puzzle/src/ctrl.ts

```diff
@@ -347,8 +347,7 @@ export default function (opts: PuzzleOpts, redraw: Redraw): Controller {
     }
     redraw();
     if (!next) {
-      alert('No more puzzles available! Try another theme.');
```

Comment:
This not only remove the alert when replaying from the puzzle dashboard, but also when playing a specific theme such as https://lichess.org/training/xRayAttack and no more puzzles are found.
The proper fix would be to detected if the user is replaying a puzzle and do not display the alert only in this case.

```ts
If (!data.replay) {
 alert(...)
}
```

Reply:
Thank you for your feedback @kraktus  I'll take care of it

Reply:
Added the alert back within an if block as suggested @kraktus